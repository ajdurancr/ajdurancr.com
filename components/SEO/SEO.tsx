import React from 'react'
import Head from 'next/head'

const SEO: React.FC<{ title?: string, siteName: string }> = ({ title = '', siteName }) => (
  <Head>
    <title>{`${title && `${title} | `}${siteName}`}</title>
  </Head>
)

export { SEO }
