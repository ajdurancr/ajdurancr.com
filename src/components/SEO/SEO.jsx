import React from 'react'
import Head from 'next/head'
import propTypes from 'prop-types'

const SEO = ({ title, siteName }) => (
  <Head>
    <title>{`${title && `${title} - `}${siteName}`}</title>
  </Head>
)


SEO.defaultProps = {
  title: '',
}

SEO.propTypes = {
  title: propTypes.string,
  siteName: propTypes.string.isRequired,
}


export default SEO
