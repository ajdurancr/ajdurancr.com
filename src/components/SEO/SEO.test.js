import React from 'react'
import { shallow } from 'enzyme'

import SEO from './SEO'

describe('SEO', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<SEO siteName="ajdurancr" />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with page title', () => {
    const wrapper = shallow(<SEO siteName="ajdurancr" title="Welcome" />)

    expect(wrapper).toMatchSnapshot()
  })
})
