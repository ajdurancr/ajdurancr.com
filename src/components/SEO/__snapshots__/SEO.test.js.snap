// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEO renders correctly 1`] = `
<Head>
  <title>
    ajdurancr
  </title>
</Head>
`;

exports[`SEO renders correctly with page title 1`] = `
<Head>
  <title>
    Welcome - ajdurancr
  </title>
</Head>
`;
